### PR TITLE
DR-1846 Set up and demonstrate ability to deploy a TDR managed app from Az Marketplace

### PR DESCRIPTION
**Pull request by nmalfroy**
This PR:
- Includes the ARM templates to describe the TDR application and application deployment
- Includes a readme for how to deploy the application in partner center
- A connected test to do a full spike through what TDR needs to do in a first pass
  - Deploy managed application
  - Create a storage account in the the managed resource group
  - Create a file
  - Sign / access a file
  - Delete the managed application deployment

### CODE REVIEW

**Comment by tlangs** [comment]
Remember to add info to the getting started doc to get this set up locally!

**Review by snf2ye — APPROVED** [review]
Thanks for walking us through these changes last week!

File: src/test/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfigurationTest.java [hunk 1]
```diff
@@ -102,4 +254,149 @@ private BillingProfileModel createProfileModel() {
...
+    private String generateSas(DataLakeFileClient client) {
+        OffsetDateTime startTime = OffsetDateTime.now();
+        OffsetDateTime expiryTime = startTime.plusHours(1);
+        FileSystemSasPermission permission = new FileSystemSasPermission().setReadPermission(true)
+            .setWritePermission(false);
+
+        // Note: can't use user delegation since that's not supported cross tenant
+        // Note: the version is so that subdirectory sharing works...docs say it's a no-op, but it sure isn't...
+        return client.generateSas(new DataLakeServiceSasSignatureValues(expiryTime, permission)
+            .setStartTime(startTime).setVersion("2020-04-08"));
```
Comment (samanehsan) [inline src/test/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfigurationTest.java]: According to the Azure docs, if we don't set a start time the token will be valid immediately so we could take this out: https://docs.microsoft.com/en-us/azure/storage/common/storage-sas-overview (under the "Be careful with SAS start time" section). It also mentions there could be intermittent delays if you set this to the current time, so if we did want to be explicit they recommend setting this to 15min in the past.

Comment (samanehsan) [reply src/test/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfigurationTest.java]: Also where does that version date come from? Is it random?

**Review by tlangs — APPROVED** [review]
This is pretty cool! Great start for us.

File: src/test/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfigurationTest.java (hunk 1 above)
Comment (nmalfroy) [reply src/test/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfigurationTest.java]: Ah great question...that `2020-04-08` doesn't refer to a date but a version of the Sas API (which is not the latest but is compatible with certain permissions we'll need).

Good point...we could also take out the start time.

Comment (samanehsan) [reply src/test/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfigurationTest.java]: Got it! Thanks for clarifying

**Pull request merged**